analyzers/src/SonarAnalyzer.Common/Rules/UseShortCircuitingOperatorBase.cs

```diff
@@ -39,35 +38,35 @@ protected static bool IsBool(SyntaxNode node, SemanticModel semanticModel)
         }
     }
 
-    public abstract class UseShortCircuitingOperatorBase<TLanguageKindEnum, TBinaryExpression> : UseShortCircuitingOperatorBase
-        where TLanguageKindEnum : struct
+    public abstract class UseShortCircuitingOperatorBase<TSyntaxKind, TBinaryExpression> : UseShortCircuitingOperatorBase<TSyntaxKind>
+        where TSyntaxKind : struct
         where TBinaryExpression : SyntaxNode
     {
-        protected sealed override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterNodeAction(
-                GeneratedCodeRecognizer,
-                c =>
+        protected sealed override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterNodeAction(Language.GeneratedCodeRecognizer, c =>
                 {
-                    var node = (TBinaryExpression)c.Node;
-
-                    if (GetOperands(node).All(o => IsBool(o, c.SemanticModel)))
+                    if (c.Node is TBinaryExpression node
+                        && Language.Syntax.BinaryExpressionLeft(node) is { } left
+                        && Language.Syntax.BinaryExpressionRight(node) is { } right
+                        && IsBool(left, c.SemanticModel)
+                        && IsBool(right, c.SemanticModel))
                     {
+                        var extractText = c.SemanticModel.GetConstantValue(right) is { HasValue: true }
+                            || c.SemanticModel.GetSymbolInfo(right).Symbol is ILocalSymbol or IFieldSymbol or IPropertySymbol or IParameterSymbol
+                                ? string.Empty
+                                : " and extract the right operand to a variable if it should always be evaluated";
```

Comment:
An alternative would be to look for side effects in the right hand side:
* invocation (`Count() > 5`)
* increment/decrement (`i++ > 5`)
* ...

Reply:
Yes. I think there is no need to go further for the time being.
We will maybe reconsider in the future, based on community inputs on the rule.